browser/ui/tabs/split_view_tab_strip_model_adapter.cc

```diff
@@ -0,0 +1,260 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/tabs/split_view_tab_strip_model_adapter.h"
+
+#include "base/memory/weak_ptr.h"
+#include "base/task/sequenced_task_runner.h"
+#include "brave/browser/ui/tabs/features.h"
+#include "chrome/browser/ui/tabs/tab_model.h"
+#include "chrome/browser/ui/tabs/tab_strip_model_observer.h"
+#include "components/tab_groups/tab_group_id.h"
+
+SplitViewTabStripModelAdapter::SplitViewTabStripModelAdapter(
+    SplitViewBrowserData& split_view_browser_data,
+    TabStripModel* model)
+    : split_view_browser_data_(split_view_browser_data), model_(model) {
+  CHECK(base::FeatureList::IsEnabled(tabs::features::kBraveSplitView));
+  CHECK(model);
+
+  model_->AddObserver(this);
+}
+
+SplitViewTabStripModelAdapter::~SplitViewTabStripModelAdapter() = default;
+
+void SplitViewTabStripModelAdapter::MakeTiledTabsAdjacent(
+    SplitViewBrowserData::Tile tile,
+    bool move_right_tab) {
+  auto [tab1, tab2] = tile;
+  auto index1 = model_->GetIndexOfTab(tab1);
+  auto index2 = model_->GetIndexOfTab(tab2);
+
+  if (index1 + 1 == index2) {
+    // Already adjacent
+    return;
+  }
+
+  if (move_right_tab) {
+    model_->MoveWebContentsAt(index2, index1 + 1, /*select_after_move*/ false);
+  } else {
+    model_->MoveWebContentsAt(index1, index2 - 1, /*select_after_move*/ false);
+  }
+}
+
+void SplitViewTabStripModelAdapter::OnTabStripModelChanged(
+    TabStripModel* tab_strip_model,
+    const TabStripModelChange& change,
+    const TabStripSelectionChange& selection) {
```

Comment:
How about early return if there is no tile?

Reply:
That's a good idea!